controllers/users/authenticate-web.js

```diff
@@ -21,68 +18,52 @@ const authenticateWeb = async (req, res) => {
   /**
    * @type {RegisterBody}
    */
-  const users = req.body;
+  const {exchangeToken} = req.body;
 
-  const checkUser = await User.findOne({
-    where: {
-      human_id: users.human_id
-    }
-  });
+  const verifyTokenResponse = await HumanIdService.verifyExchangeToken(exchangeToken);
+  if (!verifyTokenResponse.success) {
+    return res.status(500).json({
+      status: 'error on verify exchange token',
+      code: 500
+    });
+  }
+
+  const {data} = verifyTokenResponse || {};
+  const {appUserId} = data || {};
+
+  if (!appUserId) {
+    return res.status(500).json({
+      status: 'App user id is required',
+      code: 500
+    });
+  }
+
+  const checkUser = await UsersFunction.findUserByHumanId(User, appUserId);
 
   let insertedObject = {};
 
   if (checkUser) {
     insertedObject = {
       user: checkUser,
-      anonymousUser: await UsersFunction.findAnonymousUserId(User, checkUser.dataValues.user_id, {
+      anonymousUser: await UsersFunction.findAnonymousUserId(User, checkUser.user_id, {
         raw: false
       })
     };
   } else {
-    users.username = 'auth_web_' + moment().format('YYYYMMDDHHmmssSSS');
-    users.verified_status = 'UNVERIFIED';
-    /**
-     * Inserting data to Postgres DB
-     */
-    try {
-      insertedObject = await sequelize.transaction(async (t) => {
-        const user = await UsersFunction.register(User, users, t);
-        const anonymousUser = await UsersFunction.registerAnonymous(User, user?.user_id, t);
-
-        return {
-          user,
-          anonymousUser
-        };
-      });
-    } catch (e) {
-      console.log('error on sql transaction', e);
-      return res.status(500).json({
-        status: 'error on sql transaction',
-        code: 500,
-        message: e
-      });
-    }
-    /**
-     * Inserting data to Postgres DB (END)
-     */
+    return res.status(500).json({
+      status: 'Human id is not found',
+      code: 500
+    });
   }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for checking if a user exists and handling the case where they don't is sound. However, similar to the previous comment, a 400 status code might be more appropriate when the user is not found.

```diff
-    return res.status(500).json({
+    return res.status(400).json({
       status: 'Human id is not found',
-      code: 500
+      code: 400
     });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->